gadgets/cses-status/src/lib.rs

```diff
@@ -1,24 +1,73 @@
 use std::sync::{Arc, Mutex};
 
 use cses_query::{Problem, ProblemStatus};
-use egui::{Color32, CursorIcon, RichText, Sense};
+use egui::{Color32, CursorIcon, RichText, Sense, TextEdit};
+use serde::{Deserialize, Serialize};
 use utils::{Gadget, GadgetFactory, MutexExt};
 
 mod cses_query;
 
 pub struct CSESStatusGadget {
-    problem_data: Arc<Mutex<Vec<Problem>>>,
+    problem_data: ProblemData,
+    network_runtime: utils::NetworkRuntime,
+    ask_for_id: bool,
```

Comment:
We have two `ask_for_id`, one is a field and the other is a method which renders the relevant UI part.

Why not rename this field to `should_ask_for_id` instead?